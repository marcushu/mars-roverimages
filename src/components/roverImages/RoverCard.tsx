import { FunctionComponent } from "react"
import { RoverName } from "../../roverContext"
import styles from './rovercard.module.css';

interface RoverCardProps {
  roverName: RoverName
  camera: string
  id: number
  date: string
  status: string
  imageUrl: string
}

const RoverCard: FunctionComponent<RoverCardProps> = ({ roverName, camera, id, date, status, imageUrl }) => {
  return (
    <div className={styles.container} >
      <div>
        <a href={imageUrl} target='_blank' rel="noreferrer">
          <img className={styles.imagestyle} src={imageUrl} alt="roverimage" />
        </a>
      </div>
      <div className={styles.statistics}>
        <div>
          <div className={styles.rovername}>{roverName}</div>
          <div className={styles.camera}>{camera}</div>
          <div className={styles.roverid}>id: {id}</div>
        </div>
        <div>
          <div className={styles.date}>taken: {date}</div>
          <div className={styles.status}>status: {status}</div>
        </div>
      </div>
    </div>
  );
}

export default RoverCard;