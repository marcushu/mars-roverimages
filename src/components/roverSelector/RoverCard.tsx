import { FunctionComponent, useContext } from "react";
import Context, { RoverName } from "../../roverContext";
import styles from './RoverCard.module.css';

interface RoverCardProps {
  roverName: RoverName
}

//TODO: retrieve stats from api?
const RoverCard: FunctionComponent<RoverCardProps> = ({ roverName }) => {
  const rover = useContext(Context);

  return (
    <div className={styles.container} onClick={() => rover?.setRover(roverName)}>
      {roverName === rover?.roverName && <hr />}
      <div className={styles.textbox}>
        <div className={styles.roverName}>{roverName}</div>
        <table>
          <tr>
            <td>Lauch Date</td>
            <td>11-02-2021</td>
          </tr>
          <tr>
            <td>Lauch Date</td>
            <td>2012-01-26</td>
          </tr>
          <tr>
            <td>End Date</td>
            <td>2021-03-08</td>
          </tr>
        </table>
        <table>
          <tr>
            <td>Total sols</td>
            <td>1123</td>
          </tr>
          <tr>
            <td>Image count</td>
            <td>65433</td>
          </tr>
        </table>
      </div>
      {roverName === rover?.roverName && <hr />}
    </div>
  );
}

export default RoverCard;