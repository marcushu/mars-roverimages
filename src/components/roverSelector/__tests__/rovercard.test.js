import { render, screen } from "@testing-library/react"
import RoverCard from "../RoverCard"

let rovercard = null;

afterEach(() => {
    rovercard.remove();
    rovercard = null;
});

it('should render all information', () => {
    render(<RoverCard 
        imageCount={123}
        landingDate={"1-1-1"}
        lastDate={'2-2-2'}
        launchDate={'01-01-2020'}
        roverName={"Curiosity"}
        totalSols={1000} />);

    rovercard = screen.getByText(/Curiosity/i);
    expect(rovercard).toBeInTheDocument();

    const imagecount = screen.getByRole('cell', {  name: /123/i})
    expect(imagecount).toContainHTML('<td>123</td>');
    const landingdate = screen.getByRole('cell', {  name: /1-1-1/i});
    expect(landingdate).toContainHTML('<td>1-1-1</td>');
    const lastdate = screen.getByRole('cell', {  name: /2-2-2/i});
    expect(lastdate).toContainHTML('<td>2-2-2</td>');
    const launchdate = screen.getByRole('cell', {  name: /01-01-2020/i});
    expect(launchdate).toContainHTML('<td>01-01-2020</td>');
    const totalsols = screen.getByRole('cell', {  name: /1000/i})
    expect(totalsols).toContainHTML('<td>1000</td>');
});