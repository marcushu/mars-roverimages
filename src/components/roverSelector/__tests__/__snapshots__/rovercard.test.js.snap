// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="textbox"
    >
      <div
        class="roverName"
      >
        Curiosity
      </div>
      <table>
        <tbody>
          <tr>
            <td>
              Lauch Date
            </td>
            <td>
              01-01-2020
            </td>
          </tr>
          <tr>
            <td>
              Landing Date
            </td>
            <td>
              1-1-1
            </td>
          </tr>
          <tr>
            <td>
              End Date
            </td>
            <td>
              2-2-2
            </td>
          </tr>
        </tbody>
      </table>
      <table>
        <tbody>
          <tr>
            <td>
              Total sols
            </td>
            <td>
              1000
            </td>
          </tr>
          <tr>
            <td>
              Image count
            </td>
            <td>
              123
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
